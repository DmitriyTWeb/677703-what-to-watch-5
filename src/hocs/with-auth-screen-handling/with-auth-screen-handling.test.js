import React from "react";
import renderer from "react-test-renderer";
import withAuthScreenHandling from "./with-auth-screen-handling";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';

jest.mock(`../../store/selectors/selectors`, () => ({
  getAuthorizationStatus: () => `AUTH`,
}));

const MockComponent = () => {
  return (
    <div>
    </div>
  );
};

const mockStore = configureStore([]);
const MockComponentWrapped = withAuthScreenHandling(MockComponent);

describe(`Test withAuthScreenHandling with internal Component connected to Redux`, () => {
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore({});
    store.dispatch = jest.fn();

    tree = renderer.create(
        <Provider store={store}>
          <MockComponentWrapped />
        </Provider>
    );
  });

  it(`withAuthScreenHandling is rendered correctly`, () => {
    expect(tree).toMatchSnapshot();
  });
});
