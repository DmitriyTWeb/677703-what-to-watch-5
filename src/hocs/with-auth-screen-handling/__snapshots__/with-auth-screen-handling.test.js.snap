// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test withAuthScreenHandling with internal Component connected to Redux withAuthScreenHandling is rendered correctly 1`] = `<div />`;
